R/driver_rds_remote.R

```diff
@@ -0,0 +1,199 @@
+R6_driver_rds_remote <- R6::R6Class(
+  "driver_rds_s3",
+  public = list(
+    ## TODO: things like hash_algorithm: do they belong in traits?
+    ## This needs sorting before anyone writes their own driver!
+    file_ops = NULL,
+    path = NULL,
+    compress = NULL,
+    mangle_key = NULL,
+    mangle_key_pad = NULL,
+    hash_algorithm = NULL,
+    traits = list(accept = "raw"),
+
+    initialize = function(file_ops, path, compress, mangle_key, mangle_key_pad,
+                          hash_algorithm) {
+
+      is_new <- !file_ops$object_exists(path = file.path(path, "config"))
+      file_ops$create_dir(path = path)
+      file_ops$create_dir(path = file.path(path, "data"))
+      file_ops$create_dir(path = file.path(path, "keys"))
+      file_ops$create_dir(path = file.path(path, "config"))
+
+      self$file_ops <- file_ops
+      self$path <- path
+
+      ## This is a bit of complicated dancing around to mantain
```

Comment:
If you base this off one of the other drives the backward compatibility may be less problematic

Reply:
I based this off the rds driver in storr - I wasn't sure how much of the backwards compatibility stuff is needed, so I included it - if it's not needed (which I'm guessing it's not), then I'd be up for removing it.

Reply:
yeah, I recognise it - backward compatibility is hard 😢 

I have an idea to at least centralise this, but it will probably not make it in the first round of code

Reply:
this is done now in the other prs